src/mongo/mongoConnectionStrings.ts

```diff
@@ -49,6 +49,23 @@ export async function parseMongoConnectionString(connectionString: string): Prom
     let host: string;
     let port: string;
 
+    try {
+        // Attempt to connect to determine if the username/password need to be encoded
+        await connectToMongoClient(connectionString, appendExtensionUserAgent());
+    } catch (error) {
+        const parsedError: IParsedError = parseError(error);
+        if (parsedError.message.match(/unescaped/i)) {
+            const matches: RegExpMatchArray | null = connectionString.match(/^(.*):\/\/(.*):(.*)@(.*)/);
+            if (matches) {
+                const prefix: string = matches[1];
+                const username: string = matches[2];
+                const password: string = matches[3];
+                const hostAndQuery: string = matches[4];
+                connectionString = `${prefix}://${encodeURIComponent(username)}:${encodeURIComponent(password)}@${hostAndQuery}`;
```

Comment:
Nathan's recommendation in the issue was to escape with a slash, but you're using `encodeURIComponent` which is different. Do they both work?

Reply:
I was not able to get escaping with a slash to work